public/app/plugins/datasource/testdata/channels.ts

```diff
@@ -0,0 +1,52 @@
+import { ChannelSupport, ChannelHandler } from '@grafana/data';
+import {
+  SubscribeSuccessContext,
+  UnsubscribeContext,
+  JoinLeaveContext,
+  SubscribeErrorContext,
+} from 'centrifuge/dist/centrifuge.protobuf';
```

Comment:
I understand that this experimental, so ok. But this seems like a long term issue to expose the centrifuge types in the ChannelHandler interface that we expose to plugins

Reply:
you noticed!  yes the public ones are just <any> right now -- I had excluded them in the first round, but added them here so I can work with the join/leave callbacks (when other people join/leave a channel)